module/module-registration.mjs

```diff
@@ -43,3 +49,60 @@ function registerSpellLists(manifest) {
   manifest.flags.dnd5e.spellLists.forEach(uuid => dnd5e.registry.spellLists.register(uuid));
   return "spell lists";
 }
+
+/* -------------------------------------------- */
+/*  Compendium Packs                            */
+/* -------------------------------------------- */
+
+/**
+ * Apply any changes to compendium packs during the setup hook.
+ */
+export function setupModulePacks() {
+  log("Setting Up Compendium Packs", { level: "groupCollapsed" });
+  for ( const pack of game.packs ) {
+    if ( pack.metadata.type === "Item" ) pack.applicationClass = ItemCompendium5e;
+    try {
+      const complete = setupMethods.map(m => m(pack)).filter(r => r);
+      if ( complete.length ) log(`Finished setting up ${pack.metadata.label}: ${complete.join(", ")}`);
+    } catch(err) {
+      log(`Error setting up ${pack.metadata.label}\n`, { extras: [err.message], level: "error" });
+    }
+  }
+  console.groupEnd();
+}
+
+const setupMethods = [setupPackDisplay, setupPackSorting];
+
+/* -------------------------------------------- */
+
+/**
+ * Set application based on `flags.dnd5e.display`.
+ * @param {Compendium} pack  Pack to set up.
+ * @returns {string|void}    Description of the step.
+ */
+function setupPackDisplay(pack) {
+  const display = pack.metadata.flags.display ?? pack.metadata.flags.dnd5e?.display;
+  if ( display !== "table-of-contents" ) return;
+  pack.applicationClass = TableOfContentsCompendium;
+  return "table of contents";
+}
+
+/* -------------------------------------------- */
+
+let sortingChanges;
+const debouncedUpdateSorting = foundry.utils.debounce(() =>
+  game.settings.set("core", "collectionSortingModes", sortingChanges)
+, 250);
+
+/**
+ * Set default sorting order based on `flags.dnd5e.sorting`.
+ * @param {Compendium} pack  Pack to set up.
+ * @returns {string|void}    Description of the step.
+ */
+function setupPackSorting(pack) {
+  sortingChanges ??= game.settings.get("core", "collectionSortingModes") ?? {};
+  if ( !pack.metadata.flags.dnd5e?.sorting || sortingChanges[pack.metadata.id] ) return;
+  sortingChanges[pack.metadata.id] = pack.metadata.flags.dnd5e.sorting;
+  debouncedUpdateSorting();
```

Comment:
This seems a bit awkward. Can we move it out of the generalised `setupMethods` list and instead loop over the packs at the end of `setupModulePacks` again to build up this settings object?